sonic_platform_base/chassis_base.py

```diff
@@ -197,6 +222,19 @@ def get_module(self, index):
 
         return module
 
+    def get_module_index(self, module_name):
+        """
+        Retrieves module index from the module name
+
+        Args:
+            name: A string, prefixed by SUPERVISOR, LINE-CARD or FABRIC-CARD
+            Ex. SUPERVISOR0, LINE-CARD1, FABRIC-CARD5
+
+        Returns:
+            An index of the ModuleBase object in the module_list
+        """
+        raise NotImplementedError
```

Comment:
This method seems like it is platform-independent. Vendors shouldn't need to implement this, as the logic appears as though it will be common. Why is this method added here?

Reply:
This is not fixed since we have a common module_list[] across multiple module-types. We start names with SUPERVISOR0, LINE-CARD0 etc. Also, each vendor could have different number of cards of each type based on different modular chassis types.
We need a name to index mapping. This seemed like a common API not specific to module object. So, may be similar to get_all_modules() or get_num_modules() etc.